robot/Cumulus/tests/browser/engagement_plans/add_engagement_plan_to_contact.robot

```diff
@@ -2,34 +2,72 @@
-    [tags]  unstable
-    Scroll Page To Location    0    0
-    Click More Activity Button    
-    Check Activity Tasks    ${task1}    ${sub_task}    ${task2}
+    [Documentation]                      Create an Engagement plan Template with two tasks and one subtask. And  a contact
+    ...                                  Link the  engagement plan for the contact and verify
+    ...                                  There is one engagement plan set up for the contact
+    [tags]                               W-038641                 feature:Engagements
+
+
+    Go To Page                                       Listing                            Engagement_Plan_Template__c
+    Go To Engagement Plan Page                       create
+    Wait For Locator                                 frame                              Manage Engagement Plan Template
+
+    Select Frame And Click Element                   Manage Engagement Plan Template    id    idName
+    Enter Eng Plan Values                            idName                             Automation_Plan
+    Enter Eng Plan Values                            idDesc                             This plan is created via Automation
+    Click Button                                     Add Task
+    Wait Until Page Contains                         Task 1
+    Enter Task Id and Subject                        Task 1                             Task_1
+    Click Task Button                                  1                                Add Dependent Task
+    Enter Task Id and Subject                        Task 1-1                           Sub_task_1.1
+    Click Button                                     Add Task
+    Wait Until Page Contains                         Task 2
+    Enter Task Id and Subject                        Task 2                             Task_2
+    Page Scroll To Locator                           button                             Save
+    Click Button                                     Save
+    wait until location contains                     /view
+    ${ns} =  Get NPSP Namespace Prefix
+
+    Save Current Record ID For Deletion              ${ns}Engagement_Plan_Template__c
+    Go To Page                                       Details
+    ...                                              Contact
+    ...                                              object_id=${data}[contact][Id]
+
+    Select Tab                                       Related
+    Click Related List Button                        Engagement Plans               New
+    Populate Lookup Field                            Engagement Plan Template       Automation_Plan
+    Click Modal Button                               Save
+    Wait Until Modal Is Closed
+    Current Page Should Be                           Detail                         Contact
+    Validate Related Record Count                    Engagement Plans               1
+    Go To Related Engagement Actionplans Page        ${data}[contact][Id]
```

Comment:
I am not clear on this step, maybe you can show me on our Monday call @rjanjanam

Reply:
I am navigating to related engagement plans to delete the engagement plan . As the test required to delete the engagement plan and verify the tasks and subtasks exist. I am handling the delete in a different way by navigating to the related engagements list

Reply:
ahh i see it. You are opening the related list and deleting the engagement plan from there. Sounds good